tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -195,23 +241,44 @@ export class MetricsEffects implements OnInitEffects {
     fetchInfos: CardFetchInfo[],
     experimentIds: string[]
   ) {
-    /**
-     * TODO(psybuzz): if 2 cards require the same data, we should dedupe instead of
-     * making 2 identical requests.
-     */
-    const requests: TimeSeriesRequest[] = fetchInfos.map((fetchInfo) => {
-      const {plugin, tag, runId, sample} = fetchInfo;
-      const partialRequest: TimeSeriesRequest = isSingleRunPlugin(plugin)
-        ? {plugin, tag, runId: runId!}
-        : {plugin, tag, experimentIds};
-      if (sample !== undefined) {
-        partialRequest.sample = sample;
-      }
-      return partialRequest;
-    });
-
     // Fetch and handle responses.
-    return of(requests).pipe(
+    return this.multiRunTagsToEid$.pipe(
+      take(1),
+      map((tagToEid): TimeSeriesRequest[] => {
+        const requests = fetchInfos
+          .map((fetchInfo) => {
+            const {plugin, tag, runId, sample} = fetchInfo;
+
+            if (isSingleRunPlugin(plugin)) {
+              if (!runId) {
+                return;
+              }
+              return {
+                plugin,
+                tag,
+                runId,
+                sample,
```

Comment:
This assumes that sample only exists for single run plugins. Is that guaranteed by the data model? - single run vs sampled are theoretically orthoganal considerations. (I realize in practice that there are no multi-run, sampled plugins but the old code handles this case fine so I assume that is intentional).

Reply:
I've added sample to the multi run plugin request. It shouldn't ever come up, but the typing does allow for it so it's close to free to include.